kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -154,6 +188,18 @@ class BatchJobSubmission(
     }
   }
 
+  private def monitorSubmittedApp(appId: String): Unit = {
+    info(s"Monitoring submitted $batchType batch application: $appId")
+    applicationStatus = currentApplicationState
+    if (applicationStatus.isEmpty) {
+      info("The batch application not found, assume that it has finished.")
+    } else {
+      if (applicationFailed(applicationStatus)) {
```

Comment:
nit: `else { if`  => `else if`

What about other status?

Reply:
```
    if (applicationStatus.isEmpty) {
      info("The batch application not found, assume that it has finished.")
    } else if (applicationFailed(applicationStatus)) {
      throw new RuntimeException("Batch job failed:" + applicationStatus.get.mkString(","))
    } else {
      info(s"The batch application has been submitted successfully:" +
        applicationStatus.get.mkString(","))
    }
```

should we wait the application finished if it is running and spark.yarn.waitAppCompletion=true?

Reply:
cc @ulysses-you